### PR TITLE
CNTRLPLANE-1814: Add support for setting max expiration duration for KAS service account tokens

### PR DESCRIPTION
**Pull request by muraee**
## What this PR does / why we need it:

This PR adds support for setting the maximum expiration duration for service account tokens issued by the kube-apiserver. This is essential for graceful service account signing key rotation.

During SA signing key rotation, users need to enforce a limited token lifetime to ensure that all tokens are periodically refreshed and signed with the current signing key. This allows for a graceful rotation period where old tokens are still valid but eventually expire and get re-issued with the new key.

### Changes:

1. **API Change**: Added `KubeAPIServerServiceAccountTokenMaxExpiration` annotation constant that users can set on HostedCluster or HostedControlPlane resources.

2. **Control-Plane-Operator**: When the annotation is set, the CPO passes the configured duration value to kube-apiserver as the `--service-account-max-token-expiration` flag.

### Usage Example:

```yaml
apiVersion: hypershift.openshift.io/v1beta1
kind: HostedCluster
metadata:
  name: example
  annotations:
    hypershift.openshift.io/kube-apiserver-service-account-token-max-expiration: "24h"
spec:
  # ...
```

This will configure the kube-apiserver with `--service-account-max-token-expiration=24h`.

## Which issue(s) this PR fixes:

Fixes [CNTRLPLANE-1814](https://issues.redhat.com//browse/CNTRLPLANE-1814)

## Special notes for your reviewer:

- The annotation follows the existing pattern used for other KAS configuration options (like `KubeAPIServerMaximumRequestsInFlight`, `KubeAPIServerGoAwayChance`, etc.)
- The value must be a valid Go duration string (e.g., "24h", "168h", "720h")
- Minimum value is 600s (10 minutes) per Kubernetes API server requirements
- Invalid values will cause the kube-apiserver to fail to start, which will be surfaced through HCP conditions

## Checklist:
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs.
- [x] This change includes unit tests.

🤖 Generated with [Claude Code](https://claude.com/claude-code) via `/jira:solve [CNTRLPLANE-1814](https://issues.redhat.com//browse/CNTRLPLANE-1814)`

### CODE REVIEW

**Comment by muraee** [comment]
@coderabbitai review

File: api/hypershift/v1beta1/hostedcluster_types.go [hunk 1]
```diff
@@ -393,6 +393,13 @@ const (
 	// KubeAPIServerGoAwayChance allows the --goaway-chance parameter of the kube-apiserver to be overridden from its default of 0
 	KubeAPIServerGoAwayChance = "hypershift.openshift.io/kube-apiserver-goaway-chance"
 
+	// KubeAPIServerServiceAccountTokenMaxExpiration allows setting the maximum expiration duration
+	// for service account tokens issued by the kube-apiserver. This is useful during service account
+	// signing key rotation to enforce a limited token lifetime, ensuring tokens are re-issued with
+	// the new signing key. The value must be a valid Go duration string (e.g., "24h", "168h", "720h").
+	// Minimum value is 600s (10 minutes) per Kubernetes requirements.
```
Comment (jparrill) [inline api/hypershift/v1beta1/hostedcluster_types.go:400]: How are we validating the minimum value?

File: control-plane-operator/controllers/hostedcontrolplane/v2/kas/config_test.go [hunk 2]
```diff
@@ -570,6 +570,17 @@ func TestGenerateConfig(t *testing.T) {
 				},
 			),
 		},
+		{
```
Comment (jparrill) [inline control-plane-operator/controllers/hostedcontrolplane/v2/kas/config_test.go:573]: I would add a failing test with a value under minimum allowed (E.G 9min).

File: control-plane-operator/controllers/hostedcontrolplane/v2/kas/params_test.go [hunk 3]
```diff
@@ -285,6 +285,22 @@ func TestNewConfigParams(t *testing.T) {
...
+				}
+				return hcp
+			}(),
+			expected: func(hcp *hyperv1.HostedControlPlane, featureGates []string) KubeAPIServerConfigParams {
+				params := defaultKubeAPIServerConfigParams()
+				params.FeatureGates = featureGates
+				params.ServiceAccountMaxTokenExpiration = "24h"
+				return params
+			},
+		},
```
Comment (jparrill) [inline control-plane-operator/controllers/hostedcontrolplane/v2/kas/params_test.go:303]: I would add a failing test with a value under minimum allowed (E.G 9min).

File: api/hypershift/v1beta1/hostedcluster_types.go (hunk 1 above)
Comment (muraee) [reply api/hypershift/v1beta1/hostedcluster_types.go:400]: there is no validation as all other API configs we pass to the KAS though annotations. the KAS would fail which will be reflected in the conditions.
I didn't want to break the pattern of all other fields, and introduce a validation for this one specifically. Although, it makes sense that we do that at some point

File: api/hypershift/v1beta1/hostedcluster_types.go [hunk 4]
```diff
@@ -393,6 +393,13 @@ const (
...
+	KubeAPIServerServiceAccountTokenMaxExpiration = "hypershift.openshift.io/kube-apiserver-service-account-token-max-expiration"
```
Comment (rtheis) [inline api/hypershift/v1beta1/hostedcluster_types.go:401]: We should ensure that the annotation name matches the option. The annotation uses `TokenMax` and the option is `max-token`.

Comment (muraee) [reply api/hypershift/v1beta1/hostedcluster_types.go:401]: hey @rtheis, sorry I didn't catch this before the PR merged.

Comment (rtheis) [reply api/hypershift/v1beta1/hostedcluster_types.go:401]: No worries @muraee. It's not a big deal.
